codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/error/OperationErrorGenerator.kt

```diff
@@ -25,99 +25,37 @@ import software.amazon.smithy.rust.codegen.core.rustlang.rust
-import software.amazon.smithy.rust.codegen.core.smithy.CodegenTarget
 import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
 import software.amazon.smithy.rust.codegen.core.smithy.RustSymbolProvider
 import software.amazon.smithy.rust.codegen.core.smithy.customize.Section
-import software.amazon.smithy.rust.codegen.core.smithy.transformers.eventStreamErrors
-import software.amazon.smithy.rust.codegen.core.smithy.transformers.operationErrors
 import software.amazon.smithy.rust.codegen.core.util.hasTrait
 import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
 
 /**
- * For a given Operation ([this]), return the symbol referring to the unified error. This can be used
- * if you, e.g. want to return a unified error from a function:
+ * For a given Operation ([this]), return the symbol referring to the operation error. This can be used
+ * if you, e.g. want to return an operation error from a function:
  *
  * ```kotlin
  * rustWriter.rustBlock("fn get_error() -> #T", operation.errorSymbol(symbolProvider)) {
  *     write("todo!() // function body")
  * }
  * ```
  */
-/*
-    TODO(Error): refactor when core-codegen is available
-    This is a possible future implementation:
-    ```
-    inline fun <reified T : CombinedErrorGenerator> OperationShape.errorSymbol(
-        model: Model,
-        symbolProvider: RustSymbolProvider,
-        generator: KClass<T>
-    ): RuntimeType {
-        val symbol = symbolProvider.toSymbol(this)
-        return RuntimeType.forInlineFun("${symbol.name}Error", RustModule.Error) {
-                generator.java.newInstance().render(
-                    this,
-                    model,
-                    symbolProvider,
-                    symbol,
-                    this.operationErrors(model).map { it.asStructureShape().get() }
-                )
-        }
-    }
-    ```
-    Similarly for eventStreamErrorSymbol() below
- */
-fun OperationShape.errorSymbol(
-    model: Model,
-    symbolProvider: RustSymbolProvider,
-    target: CodegenTarget,
-): RuntimeType {
-    val symbol = symbolProvider.toSymbol(this)
-    return RuntimeType.forInlineFun("${symbol.name}Error", RustModule.Error) {
-        when (target) {
-            CodegenTarget.CLIENT -> CombinedErrorGenerator(
-                model,
-                symbolProvider,
-                symbol,
-                operationErrors(model).map { it.asStructureShape().get() },
-            ).render(this)
-            CodegenTarget.SERVER -> ServerCombinedErrorGenerator(
-                model,
-                symbolProvider,
-                symbol,
-                operationErrors(model).map { it.asStructureShape().get() },
-            ).render(this)
-        }
-    }
+fun OperationShape.errorSymbol(symbolProvider: RustSymbolProvider): RuntimeType {
+    val operationSymbol = symbolProvider.toSymbol(this)
+    return RuntimeType("${RustModule.Error.fullyQualifiedPath()}::${operationSymbol.name}Error")
```

Comment:
It may be worth it to add a `toType` method to `RustModule` so that we could do this instead:
```kt
return RustModule.Error.toType().resolve("${operationSymbol.name}Error")
```

Reply:
Good idea. Fixed in 684cb580887555c88cd67b421d0a9f0cade50887.